astoria/managers/astprocd.py

```diff
@@ -92,6 +108,51 @@ async def handle_disk_removal(self, uuid: DiskUUID, disk_info: DiskInfo) -> None
             else:
                 LOGGER.warning("Disk removed, but no code lifecycle available")
 
+    async def handle_kill_request(
+        self,
+        request: UsercodeKillManagerRequest,
+    ) -> RequestResponse:
+        """Handle a request to kill running usercode."""
+        if self._lifecycle is None:
+            return RequestResponse(
+                uuid=request.uuid,
+                success=False,
+                reason="No active usercode lifecycle",
+            )
+        else:
+            LOGGER.info("Kill request received.")
+            await self._lifecycle.kill_process()
+            return RequestResponse(
+                uuid=request.uuid,
+                success=True,
+            )
```

Comment:
this feels prone to issues if it happens when the usercode status is `STARTING`, when there won't be a process to kill (so `kill_process` will succeed having done nothing) but the code will start up at some future time once the robot.zip is validated

Reply:
f468ff4